format/swf/Reader.hx

```diff
@@ -334,28 +334,34 @@ class Reader {
 	function readShapeWithStyle(ver : Int) : ShapeWithStyleData {
 		var fillStyles = readFillStyles(ver);
 		var lineStyles = readLineStyles(ver);
+		bits.reset();
+		var fillBits = bits.readBits(4);
+		var lineBits = bits.readBits(4);
 		return {
 			fillStyles: fillStyles,
 			lineStyles: lineStyles,
-			shapeRecords: readShapeRecords(ver)
+			fillBits: fillBits,
+			lineBits: lineBits,
+			shapeRecords: readShapeRecords(ver, fillBits, lineBits)
 		};
 	}
 	
 	function readShapeWithoutStyle(ver : Int) : ShapeWithoutStyleData {
+		bits.reset();
+		var fillBits = bits.readBits(4);

```

Comment:
`ShapeWithoutStyle` is a little funky. For use in `DefineFontShape`, I think `fillBits` should always be 1 and `lineBits` should be 0 (because there's only one solid fill for a font glyph).

For `DefineMorphShape`, `fillBits` and `lineBits` for the StartShape should be calculated using `Tools.minBits` from the `fillStyles` and `lineStyles`. For the EndShape, I think they're always 0 because the styles have to match the StartShape? We should verify this using JPEXS.

Probably the best thing to do would be to make `fillBits` and `lineBits` parameters to this function.

Reply:
Hi Mike,

I think I resolved this in my e7767f6 commit.

Thanks.

Andrew